From 970fdfb32fbede57f4e068a47cdeeb0269b5a47f Mon Sep 17 00:00:00 2001
From: Nate Drude <nate.d@variscite.com>
Date: Wed, 11 Jan 2023 16:19:23 -0600
Subject: [PATCH] mcimx93evk: demo_apps: ethosu_apps_rpmsg: port to
 VAR-SOM-MX93

The following changes are made:

1) Move vring buffers to < 512MB space to allow for SoMs with < 512MB.
Align with vdev0vring0 and vdev1vring0 in the device tree:
arch/arm64/boot/dts/freescale/imx93-var-som-symphony.dts

2) Change debug uart from Lpuart7 to Lpuart2. This avoids a conflict with
the ethernet regulator on the SoM and outputs the M33 messages on
Symphony J18.3 and J18.5

Signed-off-by: Nate Drude <nate.d@variscite.com>
---
 boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/board.h   | 8 ++++----
 boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/pin_mux.c | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/board.h b/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/board.h
index 12d70d01..4d85f619 100644
--- a/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/board.h
+++ b/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/board.h
@@ -23,14 +23,14 @@
  ******************************************************************************/
 
 /* The UART to use for debug messages. */
-#define BOARD_DEBUG_UART_INSTANCE   2U
+#define BOARD_DEBUG_UART_INSTANCE   7U
 #define BOARD_DEBUG_UART_BAUDRATE   115200U
 #define BOARD_DEBUG_UART_TYPE       kSerialPort_Uart
-#define BOARD_DEBUG_UART_CLOCK_ROOT kCLOCK_Root_Lpuart2
+#define BOARD_DEBUG_UART_CLOCK_ROOT kCLOCK_Root_Lpuart7
 #define BOARD_DEBUG_UART_CLK_FREQ   CLOCK_GetIpFreq(BOARD_DEBUG_UART_CLOCK_ROOT)
 
-#define VDEV0_VRING_BASE (0xA4000000U)
-#define VDEV1_VRING_BASE (0xA4010000U)
+#define VDEV0_VRING_BASE (0x87ee0000U)
+#define VDEV1_VRING_BASE (0x87ef0000U)
 
 #define LED_INIT()
 #define LED_TOGGLE()
diff --git a/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/pin_mux.c b/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/pin_mux.c
index 2d39cf69..ffcef4c9 100644
--- a/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/pin_mux.c
+++ b/boards/mcimx93evk/demo_apps/ethosu_apps_rpmsg/pin_mux.c
@@ -51,12 +51,12 @@ BOARD_InitPins:
  *
  * END ****************************************************************************************************************/
 void BOARD_InitPins(void) {                                /*!< Function assigned for the core: undefined[cm33] */
-    IOMUXC_SetPinMux(IOMUXC_PAD_UART2_RXD__LPUART2_RX, 0U);
-    IOMUXC_SetPinMux(IOMUXC_PAD_UART2_TXD__LPUART2_TX, 0U);
+    IOMUXC_SetPinMux(IOMUXC_PAD_GPIO_IO09__LPUART7_RX, 0U);
+    IOMUXC_SetPinMux(IOMUXC_PAD_GPIO_IO08__LPUART7_TX, 0U);
 
-    IOMUXC_SetPinConfig(IOMUXC_PAD_UART2_RXD__LPUART2_RX, 
+    IOMUXC_SetPinConfig(IOMUXC_PAD_GPIO_IO09__LPUART7_RX,
                         IOMUXC_PAD_PD_MASK);
-    IOMUXC_SetPinConfig(IOMUXC_PAD_UART2_TXD__LPUART2_TX, 
+    IOMUXC_SetPinConfig(IOMUXC_PAD_GPIO_IO08__LPUART7_TX,
                         IOMUXC_PAD_DSE(15U));
 }
 
-- 
2.39.0

